advice-generator.js

```diff
@@ -0,0 +1,41 @@
+import { allCharacters, randomCharacter, randomQuote } from "./characters.js";
+
+function zip(list1, list2) {
+  return list1.map((element, i) => {
+    return [element, list2[i]];
+  });
+}
+
+export function selectCharacterQuotes(userInput) {
+  let charactersSelected = [];
+  let quotesSelected = [];
+  // If any of the words in the user's input match a quote title by any character, select that quote
+  if (userInput != undefined) {
+    allCharacters.forEach((character) => {
+      const allQuotes = Object.keys(character.quotes);
+      allQuotes.forEach((quote) => {
+        if (userInput.toLowerCase().includes(quote)) {
+          charactersSelected.push(character.name);
+          quotesSelected.push(character.quotes[quote]);
+        }
+      });
+    });
+  }
+
+  //If there are fewer than 3 quotes selected, choose quotes randomly until three are selected
+  while (quotesSelected.length < 3) {
+    let characterToAdd = randomCharacter();
+    let characterName = characterToAdd.name;
+    // Make sure that only 1 quote from each character is chosen
+    while (charactersSelected.includes(characterName)) {
+      characterToAdd = randomCharacter();
+      characterName = characterToAdd.name;
+    }
+    charactersSelected.push(characterName);
+    // Choose a random quote from that character
+    quotesSelected.push(randomQuote(characterToAdd));
+  }
```

Comment:
I think this could be cleaned up a bit. Without changing the algorithm too much, you could do something like the following to DRY things up:
```js
    while (true) {
      const characterToAdd = randomCharacter();
      const characterName = characterToAdd.name;

      if (!charactersSelected.includes(characterName)) {
        charactersSelected.push(characterName);
        // Choose a random quote from that character
        quotesSelected.push(randomQuote(characterToAdd));
        break;
      }
    }
```
This does leaves open the possibility that the interior loop will be infinite (should your quote data/characters change). A different approach to the the algorithm would likely be necessary to eliminate that risk.

Reply:
Thanks, this is a good suggestion!